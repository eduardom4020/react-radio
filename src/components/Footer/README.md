# Footer
>This component is a view renders the application's footer, that is the borded bar on bottom of the screen.

### Props
These are the props accepted by the component, pay attention to the description, inputs and behaviour:

---

##### height *(string | number)*
*The height of the footer. By default it's five times status bar height.*
**example**: 
```html
<Footer height={200} /> 
```

---

##### color *(string)*
*The background color in wich the footer will be filled. You should pass to it any color code accepted by CSS, like hexadecial, rgb or the color name. By default it's grey*
**example**: 
```html
<Footer color='yellow' />
```

---

##### borderColor *(string)*
*The border color of the footer. It can be colored the same way of the props above. By default it value is #a3acbe.*
**example**: 
```html
<Footer borderColor='orange' />
```

---