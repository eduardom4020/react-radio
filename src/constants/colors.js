export const BLACK_LIKE = '#20202b';
export const DARK_GREY = '#2e2e3a';
export const MEDIUM_GREY = '#a3acbe';
export const MEDIUM_ORANGE = '#eeae61';
export const METALIC_BLACK = '#191313';