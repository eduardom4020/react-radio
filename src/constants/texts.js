export const APP_TITLE = 'STATIONS';