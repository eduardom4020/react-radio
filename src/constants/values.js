export const DEFAULT_IMAGE_BT_SIZE = 192;
export const DEFAULT_ICON_BT_SIZE = 24;
export const DEFAULT_RADIO_PLAYING = 'CURRENTLY PLAYING'