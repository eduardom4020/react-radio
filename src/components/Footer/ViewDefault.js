import styled from 'styled-components';
import { View, StatusBar } from 'react-native';
import { MEDIUM_GREY } from '~/src/constants/colors';

const FooterViewDefault = styled(View)`
    width: 100%;
    height: ${props => props.height || StatusBar.currentHeight * 5 || 0}px;
    background-color: ${props => props.color || 'grey'};
    justify-content: center;
    align-items: center;
    border-style: solid;
    border-top-width: 1px;
    border-top-color: ${props => props.borderColor || MEDIUM_GREY};
`;

export default FooterViewDefault;