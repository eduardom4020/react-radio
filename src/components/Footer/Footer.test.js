import React from 'react';
import { Text, View } from 'react-native';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { DARK_GREY } from '~/src/constants/colors';
import Footer from '~/src/components/Footer';

describe('Footer', () => {
    test('Render?', () => {
        const component = shallow(
            <Footer />
        );

        expect(toJson(
            component, {
                noKey: false,
                mode: 'deep'
            }
        )).toMatchSnapshot();
    });

    test('Render with custom color?', () => {
        const component = shallow(
            <Footer color={DARK_GREY} />
        );

        expect(toJson(
            component, {
                noKey: false,
                mode: 'deep'
            }
        )).toMatchSnapshot();
    });

    test('Render with custom height?', () => {
        const component = shallow(
            <Footer height={200} />
        );

        expect(toJson(
            component, {
                noKey: false,
                mode: 'deep'
            }
        )).toMatchSnapshot();
    });
    
    test('Render with children?', () => {
        const component = shallow(
            <Footer color={DARK_GREY} >
                <View><Text>Test1</Text></View>
                <Text>Test2</Text>
            </Footer>
        );

        expect(toJson(
            component, {
                noKey: false,
                mode: 'deep'
            }
        )).toMatchSnapshot();
    });
});