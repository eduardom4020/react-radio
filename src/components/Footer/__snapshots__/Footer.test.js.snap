// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Render with children? 1`] = `
<StyledComponent
  color="#2e2e3a"
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
    width: 100%;
    height: ",
          [Function],
          "px;
    background-color: ",
          [Function],
          ";
    justify-content: center;
    align-items: center;
    border-style: solid;
    border-top-width: 1px;
    border-top-color: ",
          [Function],
          ";
",
        ],
      },
      "displayName": "Styled(View)",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": [Function],
      "toString": [Function],
      "usesTheme": false,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  <View>
    <Text>
      Test1
    </Text>
  </View>
  <Text>
    Test2
  </Text>
</StyledComponent>
`;

exports[`Footer Render with custom color? 1`] = `
<StyledComponent
  color="#2e2e3a"
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
    width: 100%;
    height: ",
          [Function],
          "px;
    background-color: ",
          [Function],
          ";
    justify-content: center;
    align-items: center;
    border-style: solid;
    border-top-width: 1px;
    border-top-color: ",
          [Function],
          ";
",
        ],
      },
      "displayName": "Styled(View)",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": [Function],
      "toString": [Function],
      "usesTheme": false,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Footer Render with custom height? 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
    width: 100%;
    height: ",
          [Function],
          "px;
    background-color: ",
          [Function],
          ";
    justify-content: center;
    align-items: center;
    border-style: solid;
    border-top-width: 1px;
    border-top-color: ",
          [Function],
          ";
",
        ],
      },
      "displayName": "Styled(View)",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": [Function],
      "toString": [Function],
      "usesTheme": false,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  height={200}
/>
`;

exports[`Footer Render? 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
    width: 100%;
    height: ",
          [Function],
          "px;
    background-color: ",
          [Function],
          ";
    justify-content: center;
    align-items: center;
    border-style: solid;
    border-top-width: 1px;
    border-top-color: ",
          [Function],
          ";
",
        ],
      },
      "displayName": "Styled(View)",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": [Function],
      "toString": [Function],
      "usesTheme": false,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
